p34/src/biguint.rs

```diff
@@ -0,0 +1,231 @@
+        // assertion `left == right` failed
+        //   left: BigUint4096 { value: [8589934593, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] }
+        //  right: BigUint4096 { value: [18446744073709551614, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] }
+
+        let a_32 = (op1 & Self::HIGHER) >> 32;
+        let b_32 = op1 & Self::LOWER;
+
+        let c_32 = (op2 & Self::HIGHER) >> 32;
+        let d_32 = op2 & Self::LOWER;
+
+        let bd_64 = b_32 * d_32;
+        let ad_64 = a_32 * d_32;
+        let bc_64 = b_32 * c_32;
+        let ac_64 = a_32 * c_32;
+
+        let [bd_l_32, bd_h_32] = Self::split(bd_64);
+        let [ad_l_32, ad_h_32] = Self::split(ad_64);
+        let [bc_l_32, bc_h_32] = Self::split(bc_64);
+        let [ac_l_32, ac_h_32] = Self::split(ac_64);
+
+        let chunk_1 = bd_l_32;
+        // let r_2_64: u64 = (bd_h_32 + ad_l_32 + bc_l_32).try_into().unwrap();
+        let chunk_2_64 = (<u32 as Into<u64>>::into(bd_h_32))
+            + (<u32 as Into<u64>>::into(ad_l_32))
+            + (<u32 as Into<u64>>::into(bc_l_32));
+        let [chunk_2, c1] = Self::split(chunk_2_64);
+
+        let chunk_3_64: u64 = (<u32 as Into<u64>>::into(ad_h_32))
+            + (<u32 as Into<u64>>::into(bc_h_32))
+            + (<u32 as Into<u64>>::into(ac_l_32))
+            + (<u32 as Into<u64>>::into(c1));
+        // let r_3_64: u64 = (ad_h_32 + bc_h_32 + ac_l_32 + c1).try_into().unwrap();
+        let [chunk_3, c2] = Self::split(chunk_3_64);
+
+        let chunk_4 = ac_h_32 + c2;
+
+        let result_1: u64 =
+            (<u32 as Into<u64>>::into(chunk_2) << 32) + <u32 as Into<u64>>::into(chunk_1);
+        let result_2: u64 =
+            (<u32 as Into<u64>>::into(chunk_4) << 32) + <u32 as Into<u64>>::into(chunk_3);
+        // let r__2: u64 = ((r_4 << 32) + r_3).try_into().unwrap();
+
+        [result_1, result_2]
+    }
+
+    fn split(ab: u64) -> [u32; 2] {
+        let low: u32 = (ab & Self::LOWER).try_into().unwrap();
+        let high: u32 = ((ab & Self::HIGHER) >> 32).try_into().unwrap();
+
+        [low, high]
+    }
+}
+
+impl<const N: usize> Default for BigUintGeneric<N> {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+#[macro_export]
+macro_rules! build_biguint_max_64 {
+    () => {
+        BigUintGeneric::new();
+    };
+    // Define the first few values, everything else is zero
+    ($($value:expr $(,)?)+) => {
+        {
+            // let mut v: Vec<u64> = vec![];
+            // $(
+            // v.push($value);
+            // )*
+            // // TODO Question: by letting the `let variable` define the type, how may I get the length of the needed array?. How do I know if it's 64 or 92, etc?
```

Comment:
question here @newpavlov

Reply:
The easiest way will be to do something like this:
```rust
let buf = [&(
    $value,
)];
BigUintGeneric::from_slice(&buf)
```